src/core/tests/util/graph_comparator.hpp

```diff
@@ -737,6 +741,39 @@ struct Equal<std::shared_ptr<Constant>> {
         return false;
     }
 };
+
+template <>
+struct Equal<std::shared_ptr<ov::Dimension>> {
+    static bool equal_value(const std::shared_ptr<ov::Dimension>& dim1, const std::shared_ptr<ov::Dimension>& dim2) {
+        if (dim1->is_dynamic() != dim2->is_dynamic())
+            return false;
+        if (dim1->get_min_length() != dim2->get_min_length())
+            return false;
+        return (dim1->get_max_length() == dim2->get_max_length());
```

Comment:
I suppose that Dimension operator== does the same thing https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/core/dimension.hpp#L37

Reply:
Yes, you are right. Used operator== in this method.